django/db/models/expressions.py

```diff
@@ -702,7 +702,12 @@ def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=
         sql_parts = []
         params = []
         for arg in self.source_expressions:
-            arg_sql, arg_params = compiler.compile(arg)
+            try:
+                arg_sql, arg_params = compiler.compile(arg)
+            except EmptyResultSet:
+                arg_sql, arg_params = getattr(arg, 'empty_result_set_value', NotImplemented), ()
+                if arg_sql is NotImplemented:
+                    raise
```

Comment:
I thought `empty_result_set_value` was meant to be a literal or whatever you'd pass to `Value`

```suggestion
                empty_result_set_value = value = getattr(arg, 'empty_result_set_value', NotImplemented)
                if empty_result_set_value is NotImplemented:
                    raise
                arg_sql, arg_params = '%s', (empty_result_set_value,)
```

Or alternatively

```suggestion
                empty_result_set_value = getattr(arg, 'empty_result_set_value', NotImplemented), ()
                if empty_result_set_value is NotImplemented:
                    raise
                arg_sql, arg_params = compiler.compile(Value(empty_result_set_value))
```

Which allows for better placeholder and `NULL` interoperability on Oracle which might have been the reason `Query.empty_result_set_value = 'NULL'` instead of `None`

https://github.com/django/django/blob/06f5c22ea5907295f2360dc83e17b89aa84ae2fe/django/db/models/expressions.py#L754-L770